manage_breast_screening/notifications/tests/management/commands/test_send_message_batch.py

```diff
@@ -107,6 +107,35 @@ def test_handle_with_appointments_outside_schedule_window(
         assert Message.objects.count() == 0
         mock_mark_batch_as_sent.assert_not_called
 
+    @pytest.mark.django_db
+    def test_handle_with_cancelled_appointments(
+        self, mock_send_message_batch, routing_plan_id
+    ):
+        """Test that that cancelled appointments are not notified"""
+        valid_appointment = AppointmentFactory(
+            starts_at=datetime.today().replace(tzinfo=TZ_INFO)
+        )
+
+        _cancelled_appointment = AppointmentFactory(
+            starts_at=datetime.today().replace(tzinfo=TZ_INFO), status="C"
+        )
+
+        subject = Command()
+        mock_mark_batch_as_sent = MagicMock()
+        subject.mark_batch_as_sent = mock_mark_batch_as_sent
+
+        subject.handle(**{"routing_plan_id": routing_plan_id})
+
+        message_batches = MessageBatch.objects.filter(routing_plan_id=routing_plan_id)
+        assert message_batches.count() == 1
+        assert (
+            Message.objects.filter(
+                appointment=valid_appointment, batch=message_batches[0]
```

Comment:
I think we need to omit the `appointment=` condition here and just filter on messages for the batch.

Reply:
ah yeah good point, have adjusted to be more specific - also discovered that the test wouldn't run independently without setting the status code so have added that too!